src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m

```diff
@@ -0,0 +1,44 @@
+function tf = verifyRuleSyntax(ruleString)
+% try to verify if a given rule string is syntactically correct
+% USAGE:
+%    tf = verifyRuleSyntax(ruleString)
+%
+% INPUT:
+%    ruleString:    The string in rule format (i.e. x(1) | x(2) & x(3) ...)
+%um
+% OUTPUT:
+%    tf:            Whether the given string is evaluateable (i.e. the
+%                   syntax is correct
+persistent x
+
+if isempty(x)
+    % 10000 genes should be large enough for most models.
+    x = true(10000,1); 
```

Comment:
Would you be able to base this number on the model size?

Reply:
I would have to pass a model into the function, and the function as such does not need one. So I would rather not do so.